src/anet.c

```diff
@@ -413,7 +413,12 @@ int anetRead(int fd, char *buf, int count)
     while(totlen != count) {
         nread = read(fd,buf,count-totlen);
         if (nread == 0) return totlen;
-        if (nread == -1) return -1;
+        if (nread == -1) 
+        {   
+            if(errno==EINTR)
+                continue;
+            return -1;
+        }
```

Comment:
This is dead now and removed here: f657315e601d28542a1f8d79ebdc1105bf803341
No need anymore.